scripts/ucalgary-run-tests

```
@@ -0,0 +1,69 @@
+ * Run UCalgary specific Unit Tests
+ *
+ * By default, this script will run all unit tests in our sites/all/modules/ucalgary directory.
+ *
+ * On first run, this script will also setup all depencencies required for it to run.
+ *
+ * Command line options:
+ *    --clean - Clear the Drupal cache, and clean any leftover test artifacts
+ *    --execute-test <test class name> - Execute a specific unit test
+ */
+
+error_reporting(E_ALL);
+ini_set('display_errors', TRUE);
+
+chdir(realpath(dirname(__FILE__) . '/..'));
+define('DRUPAL_ROOT', getcwd());
+require_once DRUPAL_ROOT . '/includes/bootstrap.inc';
+drupal_bootstrap(DRUPAL_BOOTSTRAP_DATABASE);
+
+// Setup the test script
+if (!file_exists(__DIR__ . '/../sites/all/modules/contrib/tunit/')) {
+  print  "Setting up Test modules...\n";
+  passthru("drush pm-enable simpletest --yes");
+
+  // TUnit is required to test sites/all/modules/contrib/feeds_ex
+  // Drupal won't allow us to test *anything* unless *everything* has it's dependencies,
+  // so we'll go ahead and install it
+  passthru("drush pm-download tunit");
+  passthru("drush pm-enable tunit --yes");
+}
+
+// By default, run all tests in our UCalgary modules directory
+$test_item = "--directory " . __DIR__ . "/../sites/all/modules/ucalgary/";
+
+foreach ($argv as $i => $arg) {
+  // Cleanup
+  if ($arg === '--clean') {
+    print "* Clearing Drupal caches...\n";
+    passthru("drush cc all");
+
+    print "Cleaning up previous tests..\n";
+    passthru("php " . __DIR__ . "/run-tests.sh --clean");
+  }
+  // Run a specific test
+  elseif ($arg == '--execute-test' && isset($argv[$i + 1])) {
+    $test_item = '--execute-test ' . $argv[$i + 1];
+  }
+}
+
+print "Running tests...\n";
+passthru("php " . __DIR__ . "/run-tests.sh --verbose --color $test_item");
+
+// Print all log messages
+$messages = db_select('simpletest', 's')
+  ->fields('s', ['message_id', 'file', 'line', 'message'])
+  ->condition('status', 'pass', '!=')
+  ->orderBy('message_id', 'ASC')
```

Do we only care about seeing the errors?

---

Yup, we only want to see the errors. 

This is a messy hack to get the thrown exceptions in Simpletest to print out instead of failing silently.